src/components/RegisterForm.vue

```diff
@@ -18,58 +18,53 @@
     </v-col>
   </v-row>
 </template>
-<script lang="ts">
+<script lang="ts" setup>
+
 import {isUsernameAvailable} from "@/service/AuthenticationService";
-import {defineComponent} from "vue";
+import {ref} from "vue";
 
-export default defineComponent({
-  name: 'RegisterForm',
-  props: {
-    isLoading: {
-      type: Boolean,
-      default: false
-    }
-  },
-  data() {
-    return {
-      isFormValid: false,
-      firstname: '',
-      lastname: '',
-      email: '',
-      password: '',
-      passwordRepeat: '',
-      usernameTaken: false,
-      rules: {
-        required: (value: any) => !!value || 'Pflichtfeld.',
-        email: (value: any) => /.+@.+\..+/.test(value) || 'Bitte gültige E-Mail eingeben.',
-        min: (v: any) => v.length >= 8 || 'Mindestens 8 Zeichen.',
-        mustMatchPassword: () => this.password === this.passwordRepeat || 'Passwörter stimmen nicht überein.',
-        requiresMixedCase: (value: any) => /[A-Z]/.test(value) && /[a-z]/.test(value) || 'Mindestens einen Kleinbuchstaben und einen Großbuchstaben enthalten.',
-        requiresNumber: (value: any) => /\d/.test(value) || 'Mindestens eine Zahl enthalten.',
-      }
-    }
-  },
-  methods: {
-    validateUsername() {
-      isUsernameAvailable(this.email).then((response) => {
-        if (response.data) {
-          this.usernameTaken = false;
-        } else {
-          this.usernameTaken = true;
-        }
-      });
-    },
-    onSubmit() {
-      const registerRequest = {
-        firstname: this.firstname,
-        lastname: this.lastname,
-        username: this.email,
-        password: this.password,
-      };
-      this.$emit('register', registerRequest);
+const isLoading = ref(false);
```

Comment:
isLoading war vorher ein `props`-Attribut.
Diese Attribute werden genutzt um Variablen von der Parent-Komponente in die Child-Componente zu reichen.
Wenn z.B. in LoginForm.vue (child) isLoading als prop definiert ist, kann in der Login.vue (parent) Komponente der Form folgendermaßen aufgerufen werden:

`<login-form :isLoading="...">`.

Hier müsstest du in der Composition-API die `defineProps(['foo'])`-Methode aufrufen. [Mehr Informationen](https://vuejs.org/guide/components/props.html)